examples/csplib/prob001_car_sequence.py

```diff
@@ -0,0 +1,90 @@
+"""
+Car sequencing in CPMpy (prob001 in CSPlib)
+https://www.csplib.org/Problems/prob001/
+
+Based on the Minizinc model car.mzn.
+
+Data format compatible with both variations of model (with and without block constraints)
+Model was created by Ignace Bleukx.
+"""
+import sys
+import json
+import requests
+
+import numpy as np
+from numpy.lib.stride_tricks import sliding_window_view
+
+from cpmpy import *
+
+
+
+def car_sequence(n_cars, n_options, n_classes, n_cars_p_class, options, capacity=None, block_size=None, **kwargs):
+
+  # build model
+  model = Model()
+
+  # decision variables
+  slots = intvar(0, n_classes-1, shape=n_cars, name="slots")
+  setup = boolvar(shape=(n_cars, n_options), name="setup")
+
+  # convert options to cpm_array
+  options = cpm_array(options)
+
+  # satisfy demand
+  model += [sum(slots == c) == n_cars_p_class[c] for c in range(n_classes)]
+
+  # car has correct options
+  # This can be written cleaner, see issue #117 on github
+  # m += [setup[s] == options[slots[s]] for s in range(n_cars)]
+  for s in range(n_cars):
+    model += [setup[s,o] == options[slots[s],o] for o in range(n_options)]
+
+  if capacity is not None:
+    # satisfy block capacity
+    for o in range(n_options):
+      setup_seq = setup[:,o]
+      blocks = sliding_window_view(setup_seq, block_size[o])
+      for block in blocks:
+        model += sum(block) <= capacity[o]
+
+
+  return model, (slots, setup)
+
+
+def get_data(data, pname):
+  for entry in data:
+    if pname in entry["name"]:
+      return entry
+
+
+if __name__ == "__main__":
+  # get data
+  fname = "https://raw.githubusercontent.com/CPMpy/cpmpy/csplib/examples/csplib/prob001_car_sequence.json"
```

Comment:
We want to be able to run these exact files outside the repo as well. Therefore, they are dynamically loaded. The scripts are so set up that you can also pass your own instance file as an argument

Reply:
This makes me think, maybe we should improve the argument parsing bit of the files.
@tias what do you think? We could add arguments using the `argparse` python lib so there would also be some help info for the arguments
https://docs.python.org/3/library/argparse.html

The problem description can then also be displayed when executing `prob001_car_sequence.py --help`

Reply:
Adding argparse to CSPlib examples seems rather elaborate to me, as the key goal is showing how to model CSPlib problems in CPMpy.
(most CSPlib models either hardcode one, or leave it to the user to know how to add data to a model file)

That said, I don't oppose, but if there is too much scaffolding code I fear it would be distracting.

Reply:
Yes that's true. That said, there is now some annoying and (imho) ugly code needed to parse non-default arguments anyway... 
I'll try to add the parsing to car_sequence in particular and see how much extra code it needs